semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/DocumentOps.scala

```diff
@@ -193,18 +193,28 @@ trait DocumentOps { self: DatabaseOps =>
 
               def saveDenotation(): Unit = {
                 def add(ms: m.Symbol, gs: g.Symbol): Unit = {
-                  denotations(ms) = gs.toDenotation(saveOverrides = mtree.isDefinition)
-                  if (config.overrides.isAll && mtree.isDefinition) {
-                    gs.overridesMembers.foreach {
-                      case (s, d) => denotations(s) = d
+                  val saveOverrides = config.overrides.isAll && mtree.isDefinition
+                  val DenotationResult(denot, todoOverrides, todoTpe1) =
+                    gs.toDenotation(saveOverrides)
+                  denotations(ms) = denot
+                  val todoTpe2 = todoOverrides.flatMap { ogs =>
+                    val DenotationResult(odenot, _, otodoTpe) =
+                      ogs.toDenotation(saveOverrides = true)
+                    denotations(ogs.toSemantic) = odenot
+                    otodoTpe
+                  }
+                  (todoTpe1 ++ todoTpe2).foreach { tgs =>
+                    if (tgs.isSemanticdbLocal) {
+                      val tms = tgs.toSemantic
+                      assert(tms != m.Symbol.None)
+                      if (!denotations.contains(tms)) add(tms, tgs)
                     }
                   }
                 }
-
                 if (!gsym.isOverloaded && gsym != g.definitions.RepeatedParamClass) {
                   add(symbol, gsym)
                 }
-                if (gsym.isClass && !gsym.isTrait) {
+                if (gsym.isClass && !gsym.isTrait && mtree.isDefinition) {
```

Comment:
@olafurpg No idea why this wasn't here before. Without this check, we were polluting our symbol table with useless symbols like `Int.<init>` or `Unit.<init>`.